crates/assists/src/handlers/invert_if.rs

```diff
@@ -68,6 +68,15 @@ mod tests {
 
     use crate::tests::{check_assist, check_assist_not_applicable};
 
+    #[test]
+    fn invert_if_composite_condition() {
+        check_assist(
+            invert_if,
+            "fn f() { i<|>f x == 3 || x == 4 || x == 5 { 1 } else { 3 * 2 } }",
+            "fn f() { if !(x == 3 || x == 4 || x == 5) { 3 * 2 } else { 1 } }",
```

Comment:
Looking at this, I wonder if this ideally should be de-morganed? `if x != 3 && x != 4 && x != 5`?

Reply:
I think whether de-morganing is desired depends heavily on the situation. One could argue that, when most (more than half?) of the parts of a composite condition are negative (`if x != 3 && x != 4 && y == 5`), it would be better to de-morgan (`if x== 3 || x == 4 && y != 5`). The situation in this test might also benefit from de-morganing, as it makes it clearer that this is a "x not in [3, 4, 5]" case. There might be other cases where de-morganing might make the code worse though. And in the end it also depends on the user's preferences.

I think it would be good to support it, but I'm not sure about what the best defaults are. I think adding parens might be less surprising to the user and automatic de-morganing could be opt-in through a config option, while manual de-morganing can be achieved through a separate code action. Not sure if that conforms to being the "least surprising" and "enable all features for better discovery" principles though.

Reply:
Hm, yeah, agree with the reasoning that distributing `!` should be a separate action. We actually have it, but it it is a bit naive:

```rust
fn foo() {
    if !(x == 3 |<|>| y == 5 || z == 6) {}
}

fn foo() {
    if !(!(x != 3 && y != 5) || z == 6) {}
}
```

Reply:
In the case you cite, because you invoke the assist on the `&&` operator, the current behaviour is what I would expect to happen. It would be nice if, when invoked on the prefixed `!`, the `!` could be distributed over the entire  parenthesized expression. Maybe the assist could even trigger on the `if` keyword, to allow applying De Morgan's law on the entire condition without the need for parentheses or losing the current behaviour.

Reply:
Another thing that would be nice to have is if applying De Morgan's law twice would result in the initial expression:

Currently:
`if x == 4 || y == 5 {}` -> `if !(x != 3 && y != 4) {}` -> `if !(!(x == 3 || y ==4)) {}`

That simplification is something we would probably want to do in the invert-if assist as well.